presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestAggregations.java

```diff
@@ -78,6 +78,13 @@ public void testCountWithNullIfPredicate()
         assertQuery("SELECT COUNT(*) FROM orders WHERE NULLIF(orderstatus, 'F') = orderstatus ");
     }
 
+    @Test
+    public void testAggregationPushdownThroughOuterJoinNotFiringInCorrelatedAggregates()
+    {
+        assertQuery("SELECT max(x) from (SELECT * from (VALUES (1)) t(x) LEFT JOIN (VALUES (1)) t2(y) ON t.x = t2.y) GROUP BY x",
```

Comment:
can you also test:
```
SELECT max(y) from (SELECT * from (VALUES (1)) t(x) LEFT JOIN (VALUES (1)) t2(y) ON t.x = t2.y) GROUP BY y
```

Reply:
can you add more rows to left and right join side? Maybe `1, 2, null` vs `1, 3, null`

Reply:
`(VALUES (1))` can be written as `(VALUES 1)`. Multiple single column rows can be written as `VALUES 1,2,3`.